interpreter/valid/valid.ml

```diff
@@ -95,7 +95,15 @@ let peek i (ell, ts) =
 (* Type Synthesis *)
 
 let type_value = Values.type_of
-let type_unop = Values.type_of
+let type_unop at = function
+    | Values.V128 unop ->
+      let open V128Op in
+      (match unop with
+      | I32x4ExtractLane _ -> V128Type, I32Type
```

Comment:
The idea of the unop/binop/relop/testop abstraction was that they are only used for homogeneously typed operators of the respective kind. That assumption shows up in a couple of places. Extract operations are different, so like for conversions, it's better to introduce a separate AST node for them.

Reply:
In 905d249, introduced a new ast node called ExtractLane. I put this outside of the whole Values.op type, and directly deal with it in the eval.ml, which calls into the SimdOp module logic in eval_numeric, since it knows how to evaluate other simd stuff. LMK if this approach makes sense at all. Thanks for your review!